File: PR Kelas/18 mar/Bus.cpp
#include "Bus.hpp"

Bus::Bus() : Kendaraan() {
    this->kategori="bus";
    this->kapasitas=0;
}

Bus::Bus(int nomor, int tahun_keluaran, string merk, string kategori="bus", int kapasitas) : Kendaraan(nomor, tahun_keluaran, merk, kategori) {
    this->kapasitas=kapasitas;    
}

Bus::Bus(const Bus &b) : Kendaraan(b) {
    this->kapasitas=b.kapasitas;
}

Bus& Bus::operator=(const Bus& b) {
    if (this != &b) {
        this->nomor=b.nomor;
        this->tahun_keluaran=b.tahun_keluaran;
        this->merk=b.merk;
        this->kategori=b.kategori;
        this->kapasitas=b.kapasitas;
    }
    return *this;
}

void Bus::printInfo() {
    Kendaraan::printInfo();
    cout << "Kapasitas: " << this->kapasitas << endl;
}

void Bus::biayaSewa(int lamaSewa) {
    int hargaSewa=1000000*lamaSewa;
    cout << "Biaya sewa kendaraan ini adalah: " << hargaSewa << endl;
}

File: PR Kelas/13 mei/Bus.java
class Bus extends Kendaraan {
    private int kapasitas;
    public Bus() {
        super();
        this.kategori="bus";
        this.kapasitas=0;
    }
    public Bus(int n, int t, String m, String k, int ka) {
        super(n, t, m, k);
        this.kapasitas=ka;
    }
    public Bus(Bus k) {
        this.nomor=k.nomor;
        this.tahun_keluaran=k.tahun_keluaran;
        this.merk=k.merk;
        this.kategori=k.kategori;
        this.kapasitas=k.kapasitas;
    }
    // Tidak ada operator overloading
    @Override
    public void printInfo() {
        super.printInfo();
        System.out.println("Kapasitas: " + this.kapasitas);
    }
    @Override
    public int biayaSewa(int l) {
        int hargaSewa=1000000*l;
        System.out.println("Biaya sewa kendaraan ini adalah: " + hargaSewa);
        return hargaSewa;
    }
}